includes/components/class-listing.php

```diff
@@ -78,6 +81,26 @@ public function __construct( $args = [] ) {
 		parent::__construct( $args );
 	}
 
+	/**
+	* Alters listing block settings.
+	*
+	* @param array $meta Block meta.
+	* @return array
+	*/
+	public function alter_listing_block( $meta ) {
```

Comment:
Please try to override block settings at the latest possible execution point, probably here https://github.com/hivepress/hivepress/blob/master/includes/components/class-editor.php#L128 Maybe attributes are already available when get_block is used in the code. It's ok to add the code there, we don't need to add attribute fields to the initial block settings, just in a place where they are needed.

The rest of the implementation seems clear (adding fields from the attribute Search context to the block settings, for blocks based on attribute-enabled models like Listings, Vendors, etc., then making sure these filters are applied to the block query in corresponding block implementations) let me know if there are any questions.

Reply:
Unfortunately, there is also attributes are not visible. Also, I have tried to do it here https://github.com/hivepress/hivepress/blob/93efa67a37bb05d62febc486f7d74e0139cdcc7d/includes/components/class-editor.php#L165 but it is the same result. It seems like it is needed new hook or another solution as you have said before

Reply:
Please check if there's at least a single workaround, e.g. what's the reason for this - post types are not even detected at the time of registering the block settings? Maybe we can move the block registration later in the queue, or it's not possible at all. I'm thinking of a possible workaround - just getting attributes from cache (there's a cache saved in register_attributes in the Attribute component), but maybe you'll find a less hacky way - let me know.